controllers/services/monitoring/monitoring_controller_actions.go

```diff
@@ -5,44 +5,99 @@ import (
 	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	cr "github.com/opendatahub-io/opendatahub-operator/v2/pkg/componentsregistry"
 	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 )
 
+// initialize handles all pre-deployment configurations.
 func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	log := logf.FromContext(ctx)
+	// Only handle manifests setup and initial configurations
+	platform := rr.Release.Name
+	switch platform {
+	case cluster.ManagedRhoai:
+		// Only set prometheus configmap path
+		rr.Manifests = []odhtypes.ManifestInfo{
+			{
+				Path:       odhdeploy.DefaultManifestPath,
+				ContextDir: "monitoring/prometheus/apps",
+			},
+		}
+
+	default:
+		log.Info("Monitoring enabled, won't apply changes in this mode", "cluster", platform)
+	}
+
 	return nil
 }
 
-func updateStatus(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	d, ok := rr.Instance.(*serviceApi.Monitoring)
-	if !ok {
-		return errors.New("instance is not of type *services.Monitoring")
+func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	log := logf.FromContext(ctx)
+	// Map component names to their rule prefixes
+	dscList := &dscv1.DataScienceClusterList{}
+	if err := rr.Client.List(ctx, dscList); err != nil {
+		return fmt.Errorf("failed to list DSC: %w", err)
+	}
+	if len(dscList.Items) == 0 {
+		return nil
+	}
+	dsc := &dscList.Items[0]
+
+	log.Info("")
+	// Map components to their rule prefixes
+	// TODO: Handling shared components like odh-model-controller
+	componentRules := map[string]string{
+		"dashboard":            "rhods-dashboard",
+		"workbenches":          "workbenches",
+		"kueue":                "kueue",
+		"codeflare":            "codeflare",
+		"datasciencepipelines": "data-science-pipelines-operator",
+		"modelmeshserving":     "model-mesh",
+		"ray":                  "ray",
+		"trustyai":             "trustyai",
+		"kserve":               "kserve",
+		"trainingoperator":     "trainingoperator",
+		"modelregistry":        "model-registry-operator",
+		"modelcontroller":      "odh-model-controller",
 	}
 
-	// url
-	rl := routev1.RouteList{}
-	err := rr.Client.List(
-		ctx,
-		&rl,
-		client.InNamespace(rr.DSCI.Spec.Monitoring.Namespace),
-		client.MatchingLabels(map[string]string{
-			labels.PlatformPartOf: serviceApi.MonitoringServiceName,
-		}),
-	)
+	err := cr.ForEach(func(ch cr.ComponentHandler) error {
+		enabled := cr.IsManaged(ch, dsc) && dsc.Status.Phase == status.ReadySuffix
```

Comment:
i am not sure about this part "dsc.Status.Phase == status.ReadySuffix"
do we need a special case on odh-model-controller?

Reply:
To get the actual component status, a better way would be:

```go
ci := component.NewCRObject(instance)

// read the component instance to get tha actual status
err = cli.Get(ctx, client.ObjectKeyFromObject(ci), ci)
switch {
case k8serr.IsNotFound(err):
    enabled = false
case err != nil:
    enabled = false
    // TODO: error handling
default:     
    enabled = meta.IsStatusConditionTrue(ci.GetStatus().Conditions, status.ConditionTypeReady) 
}
```

Reply:
thinking a little bit more about this comment, we probably want to distinguish between enabled and ready. If I recall the old implementation, we were supposed to install the rule after the first time the component become ready, and we remove the rules of the component gets removed

Reply:
I would like also mention this issue https://github.com/red-hat-data-services/rhods-operator/pull/316 which was from my opinion hidden by changing of timings.